kolena/dataset/dataset.py

```diff
@@ -280,13 +267,59 @@ def upload_dataset(
             assert id_fields is not None
             _upload_dataset_chunk(chunk, load_uuid, id_fields)
     assert id_fields is not None
-    request = RegisterRequest(name=name, id_fields=id_fields, uuid=load_uuid)
+    return id_fields, load_uuid
+
+
+def _send_upload_dataset_request(
+    name: str,
+    id_fields: List[str],
+    load_uuid: str,
+    sources: Optional[List[Dict[str, str]]],
+) -> EntityData:
+    request = RegisterRequest(name=name, id_fields=id_fields, uuid=load_uuid, sources=sources)
     response = krequests.post(Path.REGISTER, json=asdict(request))
     krequests.raise_for_status(response)
     data = from_dict(EntityData, response.json())
+    return data
+
+
+def _upload_dataset(
+    name: str,
+    df: Union[pd.DataFrame, Iterator[pd.DataFrame]],
+    *,
+    id_fields: Optional[List[str]] = None,
+    sources: Optional[List[Dict[str, str]]] = DEFAULT_SOURCES,
+) -> None:
+    prepared_id_fields, load_uuid = _prepare_upload_dataset_request(name, df, id_fields=id_fields)
+
+    data = _send_upload_dataset_request(name, prepared_id_fields, load_uuid, sources=sources)
```

Comment:
Are extractions of these 2 functions needed? I don't see them being shared/reused.

Reply:
they are mainly for easier reading rather than reusing.

`_send_upload_x_request` is a good abstract i noticed in the evaluation, b/c it's the wrapper to call the sdk-server endpoint. `_prepare_upload_dataset_request` method might be arguable in someway even i feel like it's easier to track.